### ISSUE TITLE
Code Insights: Add remove from dashboard insight action item

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#22870]
### Context

**TLDR;** We have to add remove insight from the dashboard action item in the insight context menu.

For the 3.30 we cut this off but it is still a very important UX moment for the code insights dashboard flow.

At the moment we have a workaround for it (you can use add insight to dashboard modal UI and uncheck insight checkbox - remove this insight from the dashboard)

<img width="343" alt="image" src="https://user-images.githubusercontent.com/18492575/125780047-371c2e53-0dec-4727-9a3d-bb426dc6713b.png">

**Tech details:**

We have to disable or remove this action item from the context menu when we interact with insight through built-in (User, Org, or Global like dashboard) or virtual (All insights) dashboards. Cause of that we have to come up with some permissions system for the insight model.

**Comment by felixfbecker** [comment]
Is this only about "Remove from this dashboard", or also about "Add to dashboard"?

**Comment by vovakulikov** [comment]
It's about `Remove from this dashboard` only. Add to the dashboard context item will be implemented in a separate issue since these two operations are not the same thing. I mean for the Add to the dashboard context item we have to implement some UI modal.

**Comment by vovakulikov** [comment]
This has been closed by this PR https://github.com/sourcegraph/sourcegraph/pull/32327
